src/centralstore/incremental/NodeManager.cpp

```diff
@@ -0,0 +1,179 @@
+#include "NodeManager.h"
+
+#include <sys/stat.h>
+
+#include "NodeBlock.h"  // To setup node DB
+#include "PropertyLink.h"
+
+NodeManager::NodeManager(std::string mode) {
+    std::ios_base::openmode openMode = std::ios::trunc;  // default is Trunc mode which overrides the entire file
+    if (mode == "app") {
+        openMode = std::ios::app;  // if app, open in append mode
+        this->nodeIndex = readNodeIndex();
+    }
+    NodeBlock::nodesDB =
+        new std::fstream(NodeManager::NODE_DB_PATH, std::ios::in | std::ios::out | openMode | std::ios::binary);
+    PropertyLink::propertiesDB =
+        new std::fstream(PropertyLink::DB_PATH, std::ios::in | std::ios::out | openMode | std::ios::binary);
+    // TODO: set PropertyLink nextPropertyIndex after validating by modulus check from file number of bytes
+
+    if (dbSize(NodeManager::NODE_DB_PATH) % NodeBlock::BLOCK_SIZE != 0) {
+        std::cout << "WARNING: " << NodeManager::NODE_DB_PATH << " might be corrupted!" << std::endl;
+    }
+}
+
+std::unordered_map<std::string, unsigned int> NodeManager::readNodeIndex() {
+    std::ifstream index_db(this->index_db_loc);
+    std::cout << "DEBUG: tellg " << index_db.tellg() << std::endl;
+    std::unordered_map<std::string, unsigned int> _nodeIndex;  // temproy node index data holder
+
+    if (index_db.is_open()) {
+        int iSize = dbSize(this->index_db_loc);
+        unsigned long dataWidth = NodeManager::INDEX_KEY_SIZE + sizeof(unsigned int);
+        if (iSize % dataWidth != 0) {
+            std::cout << "ERROR: " << this->index_db_loc << " is corrupted!" << std::endl;
+            throw std::runtime_error("DB file is corrupted!");
```

Comment:
Can we put more descriptive error message than this? For example, can we indicate on which file the corruption has occurred?

``DB file is corrupted!``

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/104/commits/23322798c6fda122e79f935250c2cea59b4f4dbc

Reply:
Done